up-transport-vsomeip/src/transport.rs

```diff
@@ -0,0 +1,566 @@
+ ********************************************************************************/
+
+use crate::determine_message_type::{
+    determine_registration_type, determine_send_type, RegistrationType,
+};
+use crate::listener_registry::{
+    find_app_name, find_available_listener_id, free_listener_id, get_extern_fn,
+    insert_into_listener_id_map, map_listener_id_to_client_id, register_listener_id_with_listener,
+    release_listener_id,
+};
+use crate::transport_inner::TransportCommand;
+use crate::transport_inner::{
+    UP_CLIENT_VSOMEIP_FN_TAG_INITIALIZE_NEW_APP_INTERNAL,
+    UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL, UP_CLIENT_VSOMEIP_FN_TAG_SEND_INTERNAL,
+    UP_CLIENT_VSOMEIP_FN_TAG_UNREGISTER_LISTENER_INTERNAL, UP_CLIENT_VSOMEIP_TAG,
+};
+use crate::vsomeip_config::extract_applications;
+use crate::UPTransportVsomeip;
+use crate::{any_uuri, any_uuri_fixed_authority_id, ApplicationName};
+use async_trait::async_trait;
+use log::{error, trace, warn};
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::sync::oneshot;
+use tokio::time::timeout;
+use up_rust::{
+    ComparableListener, UAttributesValidators, UCode, UListener, UMessage, UMessageType, UStatus,
+    UTransport, UUri,
+};
+use vsomeip_sys::extern_callback_wrappers::MessageHandlerFnPtr;
+
+const UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER: &str = "register_listener";
+
+const INTERNAL_FUNCTION_TIMEOUT: u64 = 3;
+
+async fn await_internal_function(
+    function_id: &str,
+    rx: oneshot::Receiver<Result<(), UStatus>>,
+) -> Result<(), UStatus> {
+    match timeout(Duration::from_secs(INTERNAL_FUNCTION_TIMEOUT), rx).await {
+        Ok(Ok(result)) => result,
+        Ok(Err(_)) => Err(UStatus::fail_with_code(
+            UCode::INTERNAL,
+            format!(
+                "Unable to receive status back from internal function: {}",
+                function_id
+            ),
+        )),
+        Err(_) => Err(UStatus::fail_with_code(
+            UCode::DEADLINE_EXCEEDED,
+            format!(
+                "Unable to receive status back from internal function: {} within {} second window.",
+                function_id, INTERNAL_FUNCTION_TIMEOUT
+            ),
+        )),
+    }
+}
+
+async fn send_to_inner_with_status(
+    tx: &tokio::sync::mpsc::Sender<TransportCommand>,
+    transport_command: TransportCommand,
+) -> Result<(), UStatus> {
+    tx.send(transport_command).await.map_err(|e| {
+        UStatus::fail_with_code(
+            UCode::INTERNAL,
+            format!(
+                "Unable to transmit request to internal vsomeip application handler, err: {:?}",
+                e
+            ),
+        )
+    })
+}
+
+#[async_trait]
+impl UTransport for UPTransportVsomeip {
+    async fn send(&self, message: UMessage) -> Result<(), UStatus> {
+        let attributes = message.attributes.as_ref().ok_or(UStatus::fail_with_code(
+            UCode::INVALID_ARGUMENT,
+            "Invalid uAttributes",
+        ))?;
+
+        match attributes
```

Comment:
Can remove all this here and replace with [simpler version](https://github.com/eclipse-uprotocol/up-transport-mqtt5-rust/pull/4/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R555) I can crib from up-transport-mqtt-rust.

Reply:
Done